### PR TITLE
🥬 Add TrueUSDWithPoR spec

### PR DESCRIPTION
**Pull request by MaciejNadolski98**

### CODE REVIEW

File: spec/true-currencies/scripts/TrueUSDWithPoR.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+certoraRun contracts/true-currencies/tokens/TrueUSDWithPoR.sol \
+  --verify TrueUSDWithPoR:spec/true-currencies/TrueUSDWithPoR.spec
```
Comment (yuchenlintt) [inline spec/true-currencies/scripts/TrueUSDWithPoR.sh:2]:
```suggestion
  --verify TrueUSDWithPoR:spec/true-currencies/TrueUSDWithPoR.spec \
  --rule_sanity advanced
```

**Review by yuchenlintt — APPROVED** [review]
I confirmed that the spec passes by running:
```
sh ./spec/true-currencies/scripts/TrueUSDWithPoR.sh
```

And just for fun, I also added `--rule_sanity advanced` and this spec still passes. Looks like no redundant requires in `mintCanAlwaysBeCalled`!

LGTM.

In follow-up PRs, let's also write some basic unit-test style rules (check timestamps + amounts) and a rule that confirms the total supply will always be less than or equal to the amount after a mint.

File: spec/true-currencies/TrueUSDWithPoR.spec [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    assert !lastReverted;
+}
+
+rule chainReserveFeedCanBeSetByOwner() {
+    address chainReserveFeed;
+
+    env e;
+    require e.msg.sender == owner();
+    require e.msg.value == 0;
+    setChainReserveFeed@withrevert(e, chainReserveFeed);
```
Comment (Jak0bJJ) [inline spec/true-currencies/TrueUSDWithPoR.spec:31]: Seems weird to use @withrevert here

**Review by Jak0bJJ — APPROVED** [review]

**Comment by yuchenlintt** [comment]
@Jak0bJJ if you reapprove, you should have write access to this repo now so we can merge this PR.

**Review by Jak0bJJ — APPROVED** [review]

**Review by Jak0bJJ — APPROVED** [review]

**Comment by yuchenlintt** [comment]
@msieczko could you approve so we can merge this? @Jak0bJJ's approval apparently doesn't count.

**Review by msieczko — APPROVED** [review]

**Pull request merged**